tests/exporters/test_esa_cci.py

```diff
@@ -0,0 +1,77 @@
+from pathlib import Path
+from unittest.mock import patch
+from src.exporters import ESACCIExporter
+
+class TestEsaCciExporter:
+    def test_init(self):
+        data_path = Path(tmp_path / 'data')
+        e = ESACCIExporter(data_path)
+
+        assert e.raw_folder.name == 'esa_cci_landcover'
+        assert (data_path / 'raw' / 'esa_cci_landcover').exists()
+
+    @patch('os.system', autospec=True)
+    def test_checkpointing(self, mock_system, tmp_path, capsys):
+        # checks we don't redownload files
+        exporter = ESACCIExporter(tmp_path)
+
+        # setup the already downloaded file
+        test_filename = 'testy_test.nc'
+        (tmp_path / f'raw/esa_cci_landcover/{test_filename}').touch()
+
+        exporter.wget_file(test_filename)
+        captured = capsys.readouterr()
+
+        expected_stdout = f'{test_filename} already exists! Skipping\n'
+        assert captured.out == expected_stdout, \
+            f'Expected stdout to be {expected_stdout}, got {captured.out}'
+        mock_system.assert_not_called(), 'os.system was called! Should have been skipped'
+
+    @unittest.mock.patch('os.system')
+    def test_wget_file(self, mock_system, tmp_path, capsys):
+        # tests the write call made to os.system
+        exporter = ESACCIExporter(tmp_path)
+
+        url_path = 'ftp://geo10.elie.ucl.ac.be/v207/ESACCI-LC-L4-LCCS-Map-'\
+            '300m-P1Y-1992_2015-v2.0.7b.nc.zip'
+        folder = (tmp_path / 'raw' / 'esa_cci_landcover').as_posix()
+        mock_system.assert_called_once_with(
+            f'wget {url_path} -P {folder}'
+        )
+
+    def test_read_legend(self, mock_system, tmp_path, capsys):
```

Comment:
don't know if this function would even be required? the idea was to write the response of the url request as a string and then get the pandas function to read this string. Needs some more love